### ISSUE TITLE
Quicksilver is Buggy on MacOS 14 (Sonoma)

### ISSUE THREAD

**Issue by n8henrie** [quicksilver/Quicksilver#2994]
There are a fair number of really bad Quicksilver bugs on MacOS Sonoma. We are working on them, but it is a slow process.

If Quicksilver is critical to your use of MacOS, I would recommend not yet updating to Sonoma if you can help it.

If you are already on Sonoma, several of the bugs can be worked around by disabling "Superfluous visual effects" (settings -> appearance), followed by a restart of Quicksilver.

<img width="618" alt="Screenshot 2023-12-04 at 08 37 10" src="https://github.com/quicksilver/Quicksilver/assets/1234956/5b5a2049-0756-4299-b1a2-ce7a75744c1f">

If you're unable to do so (for example some people can't get past the installation screen), paste the command below into Terminal.app (followed by <kbd>return</kbd>), which has the same effect, followed by a restart of Quicksilver.

```console
defaults write com.blacktree.Quicksilver "Use Effects" -bool NO
```

Thanks for your patience while we try to sort things out.

**Comment by jtharp-ict** [comment]
Thank you for working on this, I was not able to even get through the setup on a new machine.  The Terminal command above helped quite a lot.

**Comment by sararielle** [comment]
Glad to see this isn't a sign I broke something.  (I'm having the "detached pane" issue.) I'm so grateful for the team/community's continued work on this project.  I'm a longtime user and I look forward to the continued improvement (and resolution of these issues).

**Comment by kattkieru** [comment]
Thank you for looking into this.  I just bought a new Mac and the installation process for QS was difficult.

For anyone else: if you can't click with the laptop trackpad, try an external mouse.  It's what got me to a usable state.  The trackpad makes the window jump away like crazy.

**Comment by cgarre** [comment]
Thank you for this. I struggled a lot to get through the first screen itself, it was not just buggy, it is practically unusable. Thanks for the workaround, I had uninstalled it, now got it back.

**Comment by cgarre** [comment]
> Thank you for looking into this. I just bought a new Mac and the installation process for QS was difficult.
> 
> For anyone else: if you can't click with the laptop trackpad, try an external mouse. It's what got me to a usable state. The trackpad makes the window jump away like crazy.

unfortunately nothing worked for me. Most windows use to show up and by the time I try to go to the window, it vanishes. It was a mess, with trackpad and with mouse. I thought I will use stage manager still similar issues. Somehow i figured if i put my mouse at right place i can click through the agreement, and that helped me get past 1 screen - but this is a persistent pain.

**Comment by winstonford** [comment]
Running QS 2.4.2 smoothly on (4041) Sonoma 14.2.1

Thanks a ton for the work and update.  Before the update I had problems with launcher begin off center and top right of screen.  No more.  Working as expected.

**Comment by n8henrie** [comment]
@winstonford that's great news, thanks for the report!

**Comment by wwmiller3** [comment]
Still seeing frequent crashes with 2.4.2 on Sonoma 14.2.1. Crash log indicates SIGSEGV:

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000000
Exception Codes:       0x0000000000000001, 0x0000000000000000

**Comment by acovrig** [comment]
2.3.2 is more stable for me on Ventura, I did get a crash this morning, is there a good/better place to post crash reports to help rather than become noise?

```
Crashed Thread:        5  Dispatch queue: quicksilver.qslibrarian.scanning
Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
```

I'm hoping to use my Sonoma machine more today, so may have more insight.

I presume it's a separate issue, but my biggest issue in Ventura is typing `~/doc/gi/test` to complete to `~/Documents/git/test` and it ends up getting `~`, then the `/` for `doc` gets to the `/` root path - as-if a path searcher returns too late and it falls back to `/` (using the → key instead of `/` works (`~→doc→gi→tes`), but is less convenient)

**Comment by n8henrie** [comment]
Sorry for late response.

> is there a good/better place to post crash reports to help rather than become noise?

Here on GitHub is great. Submitting the crash reports will hopefully be eventually helpful, but the backlog of issues is long.

> my biggest issue in Ventura is typing ~/doc/gi/test to complete to ~/Documents/git/test and it ends up getting ~, then the / for doc gets to the / root path

Perhaps open a separate issue, but this seems odd. `~/git/fauxmo` selects the expected directory for me. In settings -> command, what is your `Reset search after` settings? Mine is 1.5s.

**Comment by jfrench206** [comment]
Not sure if I should open a separate issue, but I've noticed that after upgrading to Sonoma, my QS Catalog entries are randomly emptied of their contents.

The two that I've noticed this for are for the Web Searches plugin, and a custom folder that I've added. At random times they seem to need to be refreshed in the QS UI, because their contents aren't in the catalog anymore (UI shows no number next to the entry).

The other odd thing I've noticed since updating to Sonoma is that sometimes a custom trigger that I've set (command-option-T to open Terminal) doesn't work until I open Quicksilver to the Triggers section. Once I've done that, it works even when QS is not activated, as expected.
